app/utils/blocks.js

```diff
@@ -106,16 +106,80 @@ export function addError(block, message, index) {
   block.warning.bubble_.setColour('red')
 }
 
-function lineWrap(message){
+function textWasChanged(fieldName, event) {
+  return event.element === 'field' && event.name === fieldName && (event.oldValue !== event.newValue)
+}
+
+function isATextChangeEvent(fieldName) {
+  return function (event) {
+    return event.blockId === this.id && textWasChanged(fieldName, event)
+  }
+}
+
+function isAnyParentBlockDisabled(block) {
+  const parent = block.parentBlock_
+  if (!parent) return false
+  return parent.disabled || isAnyParentBlockDisabled(parent)
+}
+
+function onChange(matchesEventKind, onTrigger) {
+  return function (event) {
+    if (this.disabled || isAnyParentBlockDisabled(this)) {
+      clearValidationsFor(this)
+      return
+    }
+    if (event && (event.runCode || matchesEventKind.call(this, event))) {
+      onTrigger.call(this)
+    }
+  }
+}
```

Comment:
3. Es por esa razón que acá necesitamos hacer `call` en las funciones que definimos por parámetro. `onChange` parcialmente aplicada devuelve una función que espera un event. Esa función *si* está _bindeada_ por lo que expliqué en 1. Entonces, dentro de este scope `this` es el objeto `block`, por lo cual cuando hacemos `f.call(this)` ese `this` es el bloque, y ya no va a ser más `undefined`.

Por ese motivo tuve que agregar los `calls` que parecen medio random.